api2-war/src/main/java/eu/europeana/api2/v2/service/translate/MetadataLangDetectionService.java

```diff
@@ -0,0 +1,229 @@
+package eu.europeana.api2.v2.service.translate;
+
+import eu.europeana.api.translation.client.TranslationApiClient;
+import eu.europeana.api.translation.client.exception.TranslationApiException;
+import eu.europeana.api.translation.definitions.language.Language;
+import eu.europeana.api.translation.definitions.model.LangDetectRequest;
+import eu.europeana.api2.v2.model.translate.LanguageValueFieldMap;
+import eu.europeana.corelib.definitions.edm.beans.BriefBean;
+import eu.europeana.corelib.definitions.edm.beans.FullBean;
+import eu.europeana.corelib.definitions.edm.beans.IdBean;
+import eu.europeana.corelib.definitions.edm.entity.Proxy;
+import eu.europeana.corelib.edm.utils.EdmUtils;
+import org.apache.commons.lang3.SerializationUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.util.ReflectionUtils;
+
+import java.lang.reflect.Field;
+import java.util.*;
+
+public class MetadataLangDetectionService extends BaseService {
+
+    private static final Logger LOG = LogManager.getLogger(MetadataLangDetectionService.class);
+
+    public MetadataLangDetectionService(TranslationApiClient translationApiClient) {
+        super(translationApiClient);
+    }
+
+    /**
+     * If does not match any of the languages Europeana supports or
+     * if not supported by the language detection endpoint (ie. calling the isSupported method)
+     * then there will be no hint supplied (this means that ‘mul’ is ignored)
+     *
+     * @param bean bean that extends IdBean See: {@link IdBean}
+     * @return
+     */
+    private <T extends IdBean> String getHintForLanguageDetect(T bean, boolean searchResults) {
+        List<Language> edmLanguages = LanguageDetectionUtils.getEdmLanguage(bean, searchResults);
+        if (!edmLanguages.isEmpty()) {
+            String edmLang = edmLanguages.get(0).name().toLowerCase(Locale.ROOT);
+          //  if (getTranslationApiClient().isSupported(edmLang)) { TODO figure out a way to get supported lanaguages, for now true always
```

Comment:
How about during the initialization of the client, we request the supported languages once from Translation API and then store the result?